packages/astro/src/@types/astro.ts

```diff
@@ -1922,6 +1935,46 @@ export interface AstroUserConfig {
 				origin?: boolean;
 			};
 		};
+
+		/**
+		 * @docs
+		 * @name experimental.rewriting
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.8.0
+		 * @description
+		 *
+		 * Enables the use of rewriting features in Astro pages, Endpoints and Astro middleware:
```

Comment:
```suggestion
		 * Enables a routing feature for rewriting requests in Astro pages, Endpoints and Astro middleware, giving you programmatic control over your routes. 
```
You'll have to help me with the exact wording here, but I think we should have the word "routing" or "rerouting" in here somewhere.

(Note that I don't think "rewriting features" will be parsed in the way you intend, so I'd like to avoid that phrase.)